edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/SubscriptionPattern.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Device;
+
+    public class SubscriptionPattern
+    {
+        public SubscriptionPattern(string pattern, DeviceSubscription subscription)
```

Comment:
Feels like this can be a simple map?

Reply:
Yes, but it always iterates through the entire collection it gathers from the subscription watcher instances - 'pattern' is a regular expression it needs to match, not a simple string for lookup.